src/components/step-flow/step-flow-title/step-flow-title.component.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useContext } from "react";
+import Typography from "../../typography";
+import useLocale from "../../../hooks/__internal__/useLocale";
+import StepFlowContext from "../__internal__/step-flow.context";
+import { StyledTitleFocusWrapper } from "../step-flow.style";
+
+export interface StepFlowTitleProps {
+  /** The title of the current step.  */
+  titleString: string;
+  /** Set the variant of the internal 'Typography' component which contains the title.
+   * However, despite the chosen variant the styling will always be overridden.
+   */
+  titleVariant?: "h1" | "h2";
+  /** Override the screen reader only title with any additional context or information.
+   * If not provided, only the `titleString` prop will be used.
+   */
+  screenReaderOnlyTitle?: string;
+}
+
+let noContextWarnTriggered = false;
+
+export const StepFlowTitle = ({
+  titleString,
+  titleVariant,
+  screenReaderOnlyTitle,
+}: StepFlowTitleProps) => {
+  const locale = useLocale();
+
+  const context = useContext(StepFlowContext);
+
+  const renderScreenReaderOnlyTitle = () => {
+    if (!context) {
+      /* eslint-disable no-console */
+      if (!noContextWarnTriggered) {
+        noContextWarnTriggered = true;
+        console.warn(
+          "[WARNING] Please ensure the `StepFlowTitle` sub component passed to the `StepFlow` component" +
+            " via the `title` prop. The `StepFlowTitle` component should not be standalone. In the meantime" +
+            " the screen reader only title will be filled with the necessary placeholders."
+        );
+      }
+
+      return `${locale.stepFlow.screenReaderOnlyTitle(
+        screenReaderOnlyTitle || titleString,
+        1,
+        1
+      )}`;
+    }
+    const { validatedCurrentStep, totalSteps, category } = context;
+
+    return `${locale.stepFlow.screenReaderOnlyTitle(
+      screenReaderOnlyTitle || titleString,
+      validatedCurrentStep,
+      totalSteps,
+      category
+    )}`;
+  };
+
+  const isTitleAString = context?.isTitleAString;
+
+  if (isTitleAString) {
+    return (
+      <Typography variant={context?.titleVariant} data-element="title-text">
+        <Typography
+          fontWeight="900"
+          fontSize="var(--fontSizes600)"
+          lineHeight="var(--sizing375)"
+          variant="span"
+          aria-hidden="true"
+          data-element="visible-title-text"
+        >
+          {titleString}
+        </Typography>
```

Comment:
**suggestion(non-blocking):** You could move this into it's own variable seeing as you're kind of using the same thing twice. It means in future if we need to change the styles, we do it in one place and not in two. I'm OK with either way though.

```
const TitleTypography = ({ variant, dataElement, children }) => (
  <Typography
    fontWeight="900"
    fontSize="var(--fontSizes600)"
    lineHeight="var(--sizing375)"
    variant={variant}
    aria-hidden="true"
    data-element={dataElement}
  >
    {children}
  </Typography>
);
```
and then use it like:
```
      <TitleTypography variant="span" dataElement="visible-title-text">
        {titleString}
      </TitleTypography>
```

Reply:
Even better.